service/src/main/java/gov/nasa/pds/api/engineering/elasticsearch/ElasticSearchRegistrySearchRequestBuilder.java

```diff
@@ -126,89 +122,5 @@ public SearchRequest getSearchCollectionRequest(String queryString, String keywo
-    {
-        List<String> fields = new ArrayList<String>(), values = new ArrayList<String>();
-        Map<String,List<String>> kvps = new HashMap<String,List<String>>();
-        fields.add(field);
-        values.add(value);
-        kvps.put(key, values);
-        return getQueryForKVPs (kvps, fields, es_index);
-    }
-
-    static public SearchRequest getQueryFieldsFromKVP (String key, String value, List<String> fields, String es_index, boolean term)
-    {
-        List<String> values = new ArrayList<String>();
-        Map<String,List<String>> kvps = new HashMap<String,List<String>>();
-        values.add(value);
-        kvps.put(key, values);
-        return getQueryForKVPs (kvps, fields, es_index, term);
-    }
-
-    static public SearchRequest getQueryFieldsFromKVP (String key, List<String> values, List<String> fields, String es_index)
-    {
-        Map<String,List<String>> kvps = new HashMap<String,List<String>>();
-        kvps.put(key, values);
-        return getQueryForKVPs (kvps, fields, es_index);        
-    }
-
-    static public SearchRequest getQueryFieldsFromKVP (String key, List<String> values, List<String> fields, String es_index, boolean term)
-    {
-        Map<String,List<String>> kvps = new HashMap<String,List<String>>();
-        kvps.put(key, values);
-        return getQueryForKVPs (kvps, fields, es_index, term);      
-    }
-
-    static public SearchRequest getQueryForKVPs (Map<String,List<String>> kvps, List<String> fields, String es_index)
-    {
-        return getQueryForKVPs (kvps, fields, es_index, true);
-    }
-
-    static public SearchRequest getQueryForKVPs (Map<String,List<String>> kvps, List<String> fields, String es_index, boolean term)
-    {
-    	String[] exclude = excludes (fields);
-        String[] include = fields.toArray(new String[0]);
-        
-        BoolQueryBuilder find_kvps = QueryBuilders.boolQuery();
-        SearchRequest request = new SearchRequest(es_index)
-                .source(new SearchSourceBuilder().query(find_kvps)
-                        .fetchSource(include, exclude));
 
-        log.info("****************************************");
-        log.info("****************        exclude");
-        for (String e : exclude) log.info("****************           " + e);
-        log.info("****************************************");
-        log.info("****************        include");
-        for (String i : include) log.info("****************           " + i);
-        log.info("****************************************");
-        for (Entry<String,List<String>> key : kvps.entrySet())
-        {
-            for (String value : key.getValue())
-            {
-                if (term) find_kvps.should (QueryBuilders.termQuery (key.getKey(), value));
-                else find_kvps.should (QueryBuilders.matchQuery (key.getKey(), value));
-            }
-        }
-        return request;
-    }
```

Comment:
Why move this to a new class? The infrastructure for all queries passed through these functions which made it simple to control changes to search options.

Is there a reason putting this functionality in KVPQueryBuilder more efficient?

Reply:
There were 7 getQueryFieldsFromKVP() functions with different number of arguments (2 functions had 5 arguments) recursively calling each other. It was very hard to understand what they are doing. I needed to add one more boolean argument and few more functions with the same name. I just replaced all this madness with a new builder class.

Reply:
Yes, there are a bunch of he same call because not every caller has all the answers and Java does not allow for default values. So, one ends up with a list of cascading calls that all yet another new default value. I agree that Python like default values would be nicer but this is Java.

By have lowest common denominator builders and spreading out the default values across many places, that leads to a bug and code management madness. So either way, we get to go mad.

My point remains that without robust regression testing this kind of restructuring is dangerous as it ha a high probability of resurrecting fixed bugs.

I am not comfortable with all the change and never will be as many of the bugs fixed in the last two months was removing the large number of small builders that handled default values differently. These changes are just a return to that state. 

@tloubrieu-jpl 
Without a regression check to say that any or none have returned my preference is no more valid than @tdddblog/s so I am not sure of the point of reviewing these changes anymore.